src/UserService.Business/Commands/Avatar/GetAvatarsCommand.cs

```diff
@@ -0,0 +1,131 @@
+using LT.DigitalOffice.UserService.Models.Dto.Responses.Avatar;
+using MassTransit;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+namespace LT.DigitalOffice.UserService.Business.Commands.Avatar
+{
+  public class GetAvatarsCommand : IGetAvatarsCommand
+  {
+    private readonly IAvatarRepository _avatarRepository;
+    private readonly IUserRepository _userRepository;
+    private readonly IAvatarsResponseMapper _avatarsResponseMapper;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IRequestClient<IGetImagesRequest> _rcGetImages;
+    private readonly ILogger<GetAvatarsCommand> _logger;
+
+    private List<ImageData> GetImages(List<Guid> imageIds, List<string> errors)
+    {
+      List<ImageData> result = null;
+
+      string errorMessage = "Can't get images. Please try again later.";
+      string logMessage = "Errors while getting images with ids: {Ids}.";
+
+      try
+      {
+        Response<IOperationResult<IGetImagesResponse>> getResponse = _rcGetImages.GetResponse<IOperationResult<IGetImagesResponse>>(
+          IGetImagesRequest.CreateObj(imageIds, ImageSource.User))
+          .Result;
+
+        if (getResponse.Message.IsSuccess)
+        {
+          result = getResponse.Message.Body.ImagesData;
+        }
+        else
+        {
+          string warningMessage = logMessage + "Errors: {Errors}";
+
+          _logger.LogWarning(
+            warningMessage,
+            string.Join(", ", imageIds),
+            string.Join('\n', getResponse.Message.Errors));
+
+          errors.Add(errorMessage);
+        }
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, logMessage, string.Join(", ", imageIds));
+
+        errors.Add(errorMessage);
+      }
+
+      return result;
+    }
+
+    public GetAvatarsCommand(
+      IAvatarRepository avatarRepository,
+      IUserRepository userRepository,
+      IAvatarsResponseMapper avatarsResponseMapper,
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      ILogger<GetAvatarsCommand> logger)
+    {
+      _avatarRepository = avatarRepository;
+      _userRepository = userRepository;
+      _avatarsResponseMapper = avatarsResponseMapper;
+      _httpContextAccessor = httpContextAccessor;
+      _rcGetImages = rcGetImages;
+      _logger = logger;
+    }
+
+    public OperationResultResponse<AvatarsResponse> Execute(Guid userId, bool getOnlyCurrent)
+    {
+      OperationResultResponse<AvatarsResponse> response = new();
+      DbUser dbUser = _userRepository.Get(userId);
+
+      if (dbUser == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("User was not found.");
+      }
+      else
```

Comment:
I think this method is not needed 'else'

Reply:
and below in the method

Reply:
tried not to use many outs from method, but will fix it and others, yes